File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/movies/GetMovieLists.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.repository.MoviesRepository;
import com.moovie.backend.domain.model.MovieListListing;
import org.springframework.stereotype.Service;

@Service
public class GetMovieLists {

    private final MoviesRepository moviesRepository;

    public GetMovieLists(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieListListing invoke(int page, String userId) {
        return moviesRepository.getMovieLists(page, userId);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/movies/GetMovieLists.kt
package org.mobyle.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.repository.MoviesRepository
import org.mobyle.model.MovieListListing

class GetMovieLists(private val repository: MoviesRepository) {
    operator fun invoke(page: Int, userId: String? = null): MovieListListing = runBlocking {
        repository.getMovieLists(page, userId)
    }
}
